public/js/clients/firefox.js

```diff
@@ -54,7 +56,29 @@ let APIClient = {
       line: location.line,
       column: location.column,
       condition: condition
+    }).then(([res, bpClient]) => {
+      this._bpClients[bpClient.actor] = bpClient;
+
+      // Firefox only returns `actualLocation` if it actually changed,
+      // but we want it always to exist. Format `actualLocation` if it
+      // exists, otherwise use `location`.
+      const actualLocation = res.actualLocation ? {
+        sourceId: res.actualLocation.source.actor,
+        line: res.actualLocation.line,
+        column: res.actualLocation.column
+      } : location;
+
+      return {
+        id: bpClient.actor,
+        actualLocation: actualLocation
+      };
```

Comment:
Not a bad idea, but it's a little more confusing than Source or Location because in most places a breakpoint just *is* a `Location`. The UI doesn't have any notion of a `breakpointId` like does a `sourceId`, because a `Location` is everything is needs to talk about breakpoints. At the protocol level we want an `id` and other things like `actualLocation`.

We could make a breakpoint type for the protocol though, with `id`, `location`, and `actualLocation` fields. It's clearly for the protocol, given that it has two location fields, which we can use to know if it's moved.